node/src/accountant/mod.rs

```diff
@@ -776,7 +725,7 @@ impl Accountant {
     }
 
     fn financial_statistics(&self) -> FinancialStatistics {
-        self.financial_statistics.as_ref().borrow().clone()
+        self.financial_statistics.borrow().clone()
```

Comment:
I can do as you say, but it'll raise other problems like this:

```bash
    |
603 |         let total_paid_payable = financial_statistics.total_paid_payable;
    |                                                       ^^^^^^^^^^^^^^^^^^ unknown field
```

I can fix that using `into_inner()`, but I question myself, is this complexity really worth it?

Since, it is an easy structure now, we can use `clone()` without any trouble.

In the end, I'm biased towards the ease of reading code. I'd like to hear @dnwiebe's opinion on this.

Reply:
Okay. This will be a bit more complex regarding an explanation of the alternative. I speculate that I forgot to complete the original comment I wrote and so an adequate example was not attached. 

I'm aware of a feasible alternative to your code with .clone(). Mine files lighter, will prevent cloning in the future and will not hit one's eyes as potentially unidiomatic code. 
Let me explain how you could implement it. 

This is the getter method. Ref<_,_> is a carrier provided with the RefCell and acquired by the `.borrow()` method. (Here carefully. If you happened to get a new import in accidentally and the import was `Borrow` that is not what we want. That is a trait and a different thing, ours comes from RefCell as its method. So if you try using my code snippet and get weird compilation issues look around for this scenario, that might be it. Delete the unwanted import and all should start working just fine.)

    fn financial_statistics(&self) -> Ref<'_,FinancialStatistics> {
        self.financial_statistics.borrow()
    }

Next, at the place in the production code of using this method nothing will need to be changed. However, a lot of new issues arise in the test tree. I suggest to solve it easily, simply use an additional `.clone()` attached to that function. It will give you the wanted owned instance instead of Ref<_,_>. In tests, I don't care if the cloning needs to be there but I don't like it in the production code. :man_shrugging:

Reply:
Right from start, I was never against the idea of improving optimisation and I followed my way of using `clone()` in the hope that it will increase readability. In the end, I find the second method, that you mentioned above as _balanced between the trade-off of __readability__ and __optimisation___.